module/zfs/dsl_pool.c

```diff
@@ -548,6 +559,17 @@ dsl_pool_create(spa_t *spa, nvlist_t *zplprops __attribute__((unused)),
 #endif
 	dsl_dataset_rele_flags(ds, DS_HOLD_FLAG_DECRYPT, FTAG);
 
+	if (spa_is_shared_log(spa)) {
+		dp->dp_chain_map_obj = zap_create_flags(dp->dp_meta_objset, 0,
+		    ZAP_FLAG_HASH64 | ZAP_FLAG_UINT64_KEY |
+		    ZAP_FLAG_PRE_HASHED_KEY, DMU_OTN_ZAP_METADATA,
+		    chain_map_zap_default_bs, chain_map_zap_default_ibs,
+		    DMU_OT_NONE, 0, tx);
```

Comment:
I am not sure ZAP_FLAG_PRE_HASHED_KEY is right here, since it uses only the first 64bit of the key, which cover a pool GUID, but not objsets within, so there might be plenty of collisions.  And in theory, if I remember ZAP code right, may reach a point of ZAP leaf overflow, since unlike specification we do not support leaf chaining and with identical hash we might not be able to split the leaf.